docs: add overview page for performance section (#56828)
angular/angular: adev/src/content/guide/performance/overview.md

<docs-decorative-header title="Performance" imgSrc="adev/src/assets/images/overview.svg"> <!-- markdownlint-disable-line -->
Learn about different ways you can optimize the performance of your application.
</docs-decorative-header>

One of the top priorities of any developer is ensuring that their application is as performant as possible. These guides are here to help you follow best practices for building performant applications.

That said, please note that these best practices will only take the performance of your application so far. At the end of the day, we encourage you to measure performance in order to best understand what custom optimizations are best for your application.

| Guides Types                              | Description                                                                                                |
| :---------------------------------------- | :--------------------------------------------------------------------------------------------------------- |

> | [Deferrable views](/defer)                | Defer loading of select dependencies within a template.                                                    |

Reviewer: nit: people may know this feature as "@defer", so including it into a description would be helpful for those purposes.
```suggestion
| [Deferrable views](/defer)                | Defer loading of select dependencies within a template by wrapping corresponding parts in a `@defer` block.                                                    |
```

> | [Image optimization](/image-optimization) | Use the `NgOptimizedImage`directive to adopt best practices for loading images.                            |

Reviewer: ```suggestion
| [Image optimization](/image-optimization) | Use the `NgOptimizedImage` directive to adopt best practices for loading images.                            |
```

| [Server-side rendering](/ssr)             | Learn how to leverage rendering pages on the server to improve load times.                                 |

> | [Build-time prerendering](/prerendering)  | Also known as static-side generation (SSG), is alternate rendering method to improve load times.           |

Reviewer 2: Do we miss a word here ? 
```suggestion
| [Build-time prerendering](/prerendering)  | Also known as static-side generation (SSG), is an alternate rendering method to improve load times.           |
```

> | [Hydration](/hydration)                   | A process to improve application performance by restoring server-side rendered applications on the client. |

Reviewer 2: > by restoring server-side rendered applications on the client. 
Does the phrase feel ok to you @AndrewKushnir  ?

> | [Hydration](/hydration)                   | A process to improve application performance by restoring server-side rendered applications on the client. |

Reviewer: I'd propose the phrase it a bit differently:
```suggestion
| [Hydration](/hydration)                   | A process to improve application performance by restoring its state after server-side rendering and reusing existing DOM structure as much as possible. |
```
